addons/stock/models/product.py

```diff
@@ -766,7 +766,7 @@ def write(self, vals):
             if products_changing_company:
                 # Forbid changing a product's company when quant(s) exist in another company.
                 quant = self.env['stock.quant'].sudo().search([
-                    ('location_id.usage', 'not in', ['customer', 'supplier']),
+                    ('company_id', '!=', False),
                     ('product_id', 'in', products_changing_company.product_variant_ids.ids),
                     ('company_id', '!=', vals['company_id']),
```

Comment:
I was thinking about adding the filter in the existing one about company_id 
`('company_id', 'not in', [vals['company_id'], False])`,

Sorry if that was not clear

Reply:
got it, done updated